corehq/apps/formplayer_api/smsforms/tests/test_formplayer_interface.py

```diff
@@ -0,0 +1,67 @@
+import uuid
+
+import requests_mock
+from django.conf import settings
+from django.test import TestCase
+
+from corehq.apps.formplayer_api.smsforms.api import FormplayerInterface
+from corehq.apps.formplayer_api.utils import get_formplayer_url
+from corehq.util.hmac_request import get_hmac_digest
+
+
+class TestFormplayerInterface(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.session_id = uuid.uuid4().hex
+        self.domain = uuid.uuid4().hex
+        self.user_id = uuid.uuid4().hex
+        self.interface = FormplayerInterface(self.session_id, self.domain, self.user_id)
+
+    def test_get_raw_instance(self):
+        expected_request_data = {
+            'action': 'get-instance',
+            'session-id': self.session_id,
+            'session_id': self.session_id,
+            'domain': self.domain,
+            'oneQuestionPerScreen': True,
+            'nav_mode': 'prompt',
+        }
+        self._test_request("get-instance", self.interface.get_raw_instance, expected_request_data, {})
+
+    def test_answer_question(self):
+        mock_response = {"event": {
+            "datatype": "select",
+            "choices": ["red", "green", "blue"],
+            "caption": "What's your favorite color?",
+            "type": "question",
+            "answer": None,
+            "required": 0,
+            "ix": "1",
+            "help": None
+        }}
+
+        expected_request_data = {
+            'action': 'answer',
+            'answer': 'answer1',
+            'session-id': self.session_id,
+            'session_id': self.session_id,
+            'domain': self.domain,
+            'oneQuestionPerScreen': True,
+            'nav_mode': 'prompt',
+        }
```

Comment:
nit: The order of the JSON data objects in this test is a little disorienting. The request comes first, and then the response follows, right? I think part of what is confusing here is that the sample data: the response contains a question and the request contains `answer` (normally a question comes before an answer, but a request comes before a response)... so color me confused. Maybe reverse the order of assignments to `mock_response` and `expected_request_data` in the test, especially if this will be an example for how to write more tests in the future.

Reply:
The request is providing an answer to the current question and the response is returning the next question. Make sense? I will change the order of the variable declaration though.